src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreams.java

```diff
@@ -40,6 +92,32 @@ public MtGetRecordsResult withLastSequenceNumber(String lastSequenceNumber) {
             return this;
         }
 
+        public Date getLastApproximateCreationDateTime() {
+            return lastApproximateCreationDateTime;
+        }
+
+        public void setLastApproximateCreationDateTime(Date lastApproximateCreationDateTime) {
+            this.lastApproximateCreationDateTime = lastApproximateCreationDateTime;
+        }
+
+        public MtGetRecordsResult withLastApproximateCreationDateTime(Date lastApproximateCreationDateTime) {
+            setLastApproximateCreationDateTime(lastApproximateCreationDateTime);
+            return this;
+        }
+
+        public Integer getRecordCount() {
+            return recordCount;
+        }
+
+        public void setRecordCount(Integer recordCount) {
```

Comment:
how is this different than `records.size()`, ie: can this value be derived?

Reply:
This is the number of records loaded from the underlying stream before filtering them down to tenant records.

Reply:
should we call this `scannedRecordCount` maybe?

Reply:
That's a good idea. Will change.

Reply:
Should I rename the other properties too ... e.g., `firstScannedSequenceNumber` etc.? Or maybe create a separate contained class `ScanResult` or something that contains these five properties?

Reply:
keeping this instrumentation in a `ScannedRange` sounds good

Reply:
@sbabu-salesforce addressed in latest commit. Actually made the code much cleaner overall; thanks for the feedback! Please take a look.